hack/ci/local-kvm.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# hack script for preparing GCP to run cluster-api-provider-openstack e2e
+
+set -x -o errexit -o nounset -o pipefail
+
+function cloud_init {
+  if [[ ! -f "${devstackdir}/id_rsa" ]]; then
+    ssh-keygen -t rsa -f "${devstackdir}/id_rsa" -N ''
+    chmod 600 "${devstackdir}/id_rsa"
+  fi
+
+  ubuntuImage="${devstackdir}/focal-server-cloudimg-amd64.img"
+  ubuntuImageURL="https://storage.googleapis.com/artifacts.k8s-staging-capi-openstack.appspot.com/test/ubuntu/2023-01-14/focal-server-cloudimg-amd64.img"
+  ubuntuImageSha256Sum="2aa7fb737962f00da3ac21f542dff7f8d966ee10a4e99de9422d406107e74d92"
+  if [[ ! -f "${ubuntuImage}" ]] || [[ ! "$(sha256sum "${ubuntuImage}")" =~ ${ubuntuImageSha256Sum} ]]; then
+    wget -O "${ubuntuImage}" "$ubuntuImageURL"
+  fi
+
+  KVM_VCPUS_controller=${KVM_VCPUS:-"16"}
+  KVM_VCPUS_worker=${KVM_VCPUS:-"8"}
+  KVM_MEM_controller=${KVM_MEM:-"16384"}
```

Comment:
seems my local env doesn't need such big memory :)

Reply:
And you can run all CAPO e2e tests on it? Or just a single cluster?

Reply:
oh.. I didn't run e2e, I just run a cluster..

Reply:
Okay, in this case you're fine I think. But running e2e tests results in `SHUTOFF` instances because of missing resources 😞

Reply:
Can we potentially use smaller flavors?
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/c115e7487a869f3a356fd67423a3a19122583703/hack/ci/openstack.sh#L35

Reply:
@seanschneeweiss maybe this is the issue. Let me try using smaller flavors ;)